hilti/runtime/include/logging.h

```diff
@@ -36,7 +36,8 @@ void warning(std::string_view msg);
     }
 
 /** Shortcut to `hilti::rt::debug::setLocation`. */
-#define __location__(x) ::hilti::rt::debug::detail::tls_location = x;
+// #define __location__(x) ::hilti::rt::debug::detail::tls_location = x;
```

Comment:
This is needed for location reporting for exceptions, so if we wanted to get rid of it we would need to introduce some other, cheaper mechanism.

That being said, I look into what overhead using thread-local storage introduces, and it seems moderate and on a fraction of the total cost (practically just a memory access).